### PR TITLE
avoid deep pattern for maven-invoker-plugin generated junit files

### PR DESCRIPTION
**Pull request by olamy**

### CODE REVIEW

**Comment by dduportal** [comment]
Hello 👋 Could you explain the why/what of this change? It's not clear what problem it is solving?

File: vars/buildPlugin.groovy [hunk 1]
```diff
@@ -115,7 +115,7 @@ def call(Map params = [:]) {
                                     infra.runMaven(mavenOptions, jdk, null, null, addToolEnv)
                                 } finally {
                                     if (!skipTests) {
-                                        junit('**/target/surefire-reports/**/*.xml,**/target/failsafe-reports/**/*.xml,**/target/invoker-reports/**/*.xml')
```
Comment (dduportal) [inline vars/buildPlugin.groovy:118]: If I understand correctly this change, the junit reports that would be in nested directories (for instance `target/surefire-reports/foo/test-1.xml`) would not be selected anymore.

Am I correct?

**Comment by dduportal** [comment]
`java.lang.AssertionError
	at BuildPluginStepTests.test_buildPlugin_with_defaults(BuildPluginStepTests.groovy:130)`

do not forget to update the unit test ;)

**Comment by timja** [comment]
> Hello 👋 Could you explain the why/what of this change? It's not clear what problem it is solving?

Context: https://github.com/jenkinsci/plugin-pom/pull/508

File: vars/buildPlugin.groovy (hunk 1 above)
Comment (timja) [reply vars/buildPlugin.groovy:118]: I don't think they are actually nested

**Review by jglick — COMMENTED** [review]
While I think this change is correct, I doubt it will solve the problem in https://github.com/jenkinsci/plugin-pom/pull/508 which IIUC was not about `target/surefire-reports/some/deep/subdir/TEST-Stuff.xml` but about `some/deep/subdir/target/surefire-reports/TEST-Stuff.xml`. And switching to for example `target/surefire-reports/*.xml` would cause us to lose legitimate test results from multimodule repos.

File: vars/buildPlugin.groovy (hunk 1 above)
Comment (olamy) [reply vars/buildPlugin.groovy:118]: surefire xml files are only generated at the top level directory.

**Comment by olamy** [comment]
> While I think this change is correct, I doubt it will solve the problem in [jenkinsci/plugin-pom#508](https://github.com/jenkinsci/plugin-pom/pull/508) which IIUC was not about `target/surefire-reports/some/deep/subdir/TEST-Stuff.xml` but about `some/deep/subdir/target/surefire-reports/TEST-Stuff.xml`. And switching to for example `target/surefire-reports/*.xml` would cause us to lose legitimate test results from multimodule repos.

the new pattern `**/target/surefire-reports/*.xml`  still starts with `**` so that should not be a problem with multimodule repos.

**Comment by jglick** [comment]
Right, so (AFAICT) this change should not _break_ anything, but neither is it likely to _fix_ the problem in https://github.com/jenkinsci/plugin-pom/pull/508.

**Comment by olamy** [comment]
Even if those deep scanning could be avoided, the PR is not needed anymore.

**Comment by jglick** [comment]
https://github.com/jenkinsci/plugin-pom/pull/508#issuecomment-1051313886: maybe would be appropriate to look only one level deep?

**Pull request closed without merging**